llama-index-integrations/agent/llama-index-agent-nvidia/llama_index/agent/nvidia/base.py

```diff
@@ -0,0 +1,137 @@
+    ```
+
+    """
+
+    def __init__(
+        self,
+        tools: List[BaseTool],
+        memory: BaseMemory,
+        prefix_messages: List[ChatMessage],
+        verbose: bool = False,
+        llm: Optional[NVIDIA] = None,
+        max_function_calls: int = DEFAULT_MAX_FUNCTION_CALLS,
+        default_tool_choice: str = "auto",
+        callback_manager: Optional[CallbackManager] = None,
+        tool_retriever: Optional[ObjectRetriever[BaseTool]] = None,
+        tool_call_parser: Optional[Callable[[OpenAIToolCall], Dict]] = None,
+    ) -> None:
+        """Init params."""
+        if not llm:
+            llm = NVIDIA("meta/llama-3.1-70b-instruct", is_function_calling_model=True)
+        callback_manager = callback_manager or llm.callback_manager
+        step_engine = OpenAIAgentWorker.from_tools(
+            tools=tools,
+            tool_retriever=tool_retriever,
+            llm=llm,
+            verbose=verbose,
+            max_function_calls=max_function_calls,
+            callback_manager=callback_manager,
+            prefix_messages=prefix_messages,
+            tool_call_parser=tool_call_parser,
+        )
+        super().__init__(
+            step_engine,
+            memory=memory,
+            llm=llm,
+            callback_manager=callback_manager,
+            default_tool_choice=default_tool_choice,
+        )
+
+    @classmethod
+    def from_tools(
+        cls,
+        tools: Optional[List[BaseTool]] = None,
+        tool_retriever: Optional[ObjectRetriever[BaseTool]] = None,
+        llm: Optional[NVIDIA] = None,
+        chat_history: Optional[List[ChatMessage]] = None,
+        memory: Optional[BaseMemory] = None,
+        memory_cls: Type[BaseMemory] = ChatMemoryBuffer,
+        verbose: bool = False,
+        max_function_calls: int = DEFAULT_MAX_FUNCTION_CALLS,
+        default_tool_choice: str = "auto",
+        callback_manager: Optional[CallbackManager] = None,
+        system_prompt: Optional[str] = None,
+        prefix_messages: Optional[List[ChatMessage]] = None,
+        tool_call_parser: Optional[Callable[[OpenAIToolCall], Dict]] = None,
+        **kwargs: Any,
+    ) -> "NVIDIAAgent":
+        """Create an NVIDIAAgent from a list of tools.
+
+        Similar to `from_defaults` in other classes, this method will
+        infer defaults for a variety of parameters, including the LLM,
+        if they are not specified.
+
+        """
+        tools = tools or []
+
+        chat_history = chat_history or []
+        if not llm:  # default model
+            llm = NVIDIA("meta/llama-3.1-70b-instruct", is_function_calling_model=True)
+
+        if callback_manager is not None:
+            llm.callback_manager = callback_manager
+
+        memory = memory or memory_cls.from_defaults(chat_history, llm=llm)
+
+        # only llama3 endpoints support tool calling
+        # TODO: update filtering to support future models
+        if not llm.model.startswith("meta/llama-3.1-"):
```

Comment:
lift this into a global table that can be checked.

model endpoints that currently support tool calling -
- nv-mistralai/mistral-nemo-12b-instruct
- meta/llama-3.1-8b-instruct
- meta/llama-3.1-70b-instruct
- meta/llama-3.1-405b-instruct
- mistralai/mistral-large-2-instruct